lumen/ai/agents.py

```diff
@@ -201,6 +196,21 @@ async def _select_table(self, tables):
         self.interface.pop(-1)
         return tables
 
+    def _render_prompt(self, prompt_name: str, **context) -> str:
+        context["memory"] = self._memory
+        system_prompt = render_template(
+            self.template_paths[prompt_name],
+            prompt_overrides=self.prompt_overrides,
+            **context
+        )
+        return system_prompt
+
+    @abstractmethod
+    async def _render_system_prompt(self, messages: list[Message]) -> str:
```

Comment:
By system prompt this refers to the "main" prompt?

Reply:
Yeah. I suppose I should rename to render_main_prompt